crates/bevy_ui/src/widget/viewport.rs

```diff
@@ -0,0 +1,205 @@
+use bevy_asset::{Assets, Handle};
+use bevy_ecs::{
+    bundle::Bundle,
+    children,
+    entity::Entity,
+    event::EventReader,
+    hierarchy::Children,
+    prelude::Component,
+    query::Changed,
+    reflect::ReflectComponent,
+    spawn::SpawnRelated,
+    system::{Commands, Local, Query, Res, ResMut},
+};
+use bevy_image::Image;
+use bevy_math::Rect;
+#[cfg(feature = "bevy_ui_picking_backend")]
+use bevy_picking::{
+    events::PointerState,
+    hover::HoverMap,
+    pointer::{Location, PointerId, PointerInput},
+    Pickable,
+};
+use bevy_platform_support::collections::HashSet;
+use bevy_reflect::Reflect;
+use bevy_render::{
+    camera::{Camera, NormalizedRenderTarget},
+    render_resource::Extent3d,
+};
+use bevy_transform::components::GlobalTransform;
+use bevy_utils::default;
+#[cfg(feature = "bevy_ui_picking_backend")]
+use uuid::Uuid;
+
+use crate::{ComputedNode, ImageNode, Node, PositionType, Val};
+
+/// Component used to render a [`Camera::target`]  to a node.
+///
+/// # See Also
+///
+/// [`on_add_viewport`]
+/// [`update_viewport_render_target_size`]
+#[derive(Component, Debug, Clone, Copy, Reflect)]
+#[reflect(Component, Debug)]
+pub struct Viewport {
```

Comment:
Name is confusing as there's already a `Viewport` type.
```suggestion
pub struct ViewportNode {
```